src/Bicep.LangServer.UnitTests/Handlers/BicepDecompileForPasteCommandHandlerTests.cs

```diff
@@ -0,0 +1,1150 @@
+                                // although a missing parameter at the top level would not (because the top level is
+                                //   not expected to be complete).
+                                ""location"": ""[parameters('location')]""
+                            }
+                        ]
+                    }
+                }
+            }";
+
+            await TestDecompileForPaste(
+                json: json,
+                PasteType.Resources,
+                expectedErrorMessage: "[18:48]: Unable to find parameter location",
+                expectedBicep: null);
+        }
+
+        [TestMethod]
+        public async Task SingleResource_WithNestedOuterScopedTemplate_WithMissingParam_ShouldBePastable_ButShouldGiveError()
+        {
+            const string json = @"
+            {
+                ""name"": ""nestedDeploymentInner"",
+                ""type"": ""Microsoft.Resources/deployments"",
+                ""apiVersion"": ""2021-04-01"",
+                ""properties"": {
+                    ""mode"": ""Incremental"",
+                    ""template"": {
+                        ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#"",
+                        ""contentVersion"": ""1.0.0.0"",
+                        ""resources"": [
+                            {
+                                ""name"": ""storageaccount1"",
+                                ""type"": ""Microsoft.Storage/storageAccounts"",
+                                ""apiVersion"": ""2021-04-01"",
+                                // Refers to a local parameter in the module, which is missing.
+                                // This should cause error during conversion (because the nested template should be valid),
+                                // although a missing parameter at the top level would not (because the top level is
+                                //   not expected to be complete).
+                                ""location"": ""[parameters('location')]""
+                            }
+                        ]
+                    }
+                }
+            }";
+
+            await TestDecompileForPaste(
+                json: json,
+                PasteType.Resources,
+                expectedErrorMessage: null,
+                expectedBicep: @"
+                    module nestedDeploymentInner './nested_nestedDeploymentInner.bicep' = {
+                        name: 'nestedDeploymentInner'
+                        params: {
+                            location: location
+                        }
+                    }");
+        }
+
+        [TestMethod]
+        public async Task MultipleResourceObjects_ExtraBraceAfterwards_ShouldSucceed()
+        {
+            string json = @$"
+                    {Resource1Json}
+                    {Resource2Json}
+                }}}} // extra";
```

Comment:
Can we add test case for below to document we support those scenarios as well?
```
 @$"
  {Resource1Json}
  {Resource2Json}
{}
```
And 
```
 @$"
  {Resource1Json}
  {Resource2Json}
{
```

Reply:
153b64b8f1c67fed1835d425fc2b35c630afff6a